### DISCUSSION TITLE
gdate command does not work in Bash script

### DISCUSSION THREAD

**Discussion by bikermikem** [Homebrew/discussions#144] [Everyday usage] [answered]
iMac (2017) with OS Mojave 10.14.6 (18G6020)
I've installed the gdate command using brew and it works from the interactive command line but not in a bash script.

Script:
#!/bin/bash
old_date=`grep 'Last updated' /etc/hosts | cut -c21-32 | xargs`
echo $old_date
# convert to epoch.
set -x
old_date=($(gdate  --date="$old_date" +%s))
echo $old_date

From error log:
+ old_date=($(gdate  --date="$old_date" +%s))
++ gdate '--date=29 Oct 2020' +%s
/Users/mike/Scripts/hosts_get/hosts_get.sh: line 11: gdate: command not found

From Command Line:
~/Scripts/hosts_get  -------------------------------------------------------------------------------------------  11:19:15
$ old_date=`grep 'Last updated' /etc/hosts | cut -c21-32 | xargs`
~/Scripts/hosts_get  -------------------------------------------------------------------------------------------  11:19:23
$ echo $old_date
29 Oct 2020
~/Scripts/hosts_get  -------------------------------------------------------------------------------------------  11:19:36
$ old_date=($(gdate  --date="$old_date" +%s))
~/Scripts/hosts_get  -------------------------------------------------------------------------------------------  11:20:04
$ echo $old_date
1603929600

**Comment by SMillerDev** [comment]
What does `echo $PATH` report in both environments?

**Reply by bikermikem** [reply]
$ echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Applications/Wireshark.app/Contents/MacOS:~/bin:/usr/local/Cellar/qt/5.11.1:/usr/local/opt/qt/bin:/usr/local/sbin:/usr/local/opt/gettext/bin

I'm set for bash in my user preferences, not ksh.

<img width="462" alt="Screenshot 2020-11-16 at 17 27 27" src="https://user-images.githubusercontent.com/1764753/99286659-1376e680-2831-11eb-85d7-623a09d738f5.png">

**Reply by SMillerDev** [reply]
That is what it returns in your shell, but what does it return in a script?

**Comment by bikermikem** [comment, marked as answer]
+ echo 'PATH = '
+ echo /usr/bin:/bin:/usr/sbin:/sbin

Don't understand why it doesn't pick up my path additions in .profile but thinking it might be due to running as a Global Deamon in launchctrl so running as root I believe.  Sorry I'm not a Unix guru, DEC's OpenVMS is my forte.  Wish I could run it on a Mac.

**Reply by gromgit** [reply]
Yeah, scripts run outside of a Terminal window generally don't pick up the stuff in `~/.profile`. If it's being run as root, it won't even be looking at the right user's `.profile` anyway.

Your best bet is probably to hardcode the path to `gdate` in your script. Assuming you use it multiple times in your script, add the following near the top:
```
gdate=/usr/local/bin/gdate
if [ ! -x $gdate ]; then
  echo "FATAL ERROR: $gdate not found." >&2
  exit 1
fi
```
then substitute `gdate` with `$gdate` everywhere else in your script.

**Comment by bikermikem** [comment]
Thanks, that worked.  Take Care, Mike
